### PR TITLE
issue_18 Add Ticket Filters

### LINKED ISSUE #18: Add Ticket Filters
Add filters to show tickets based on status (e.g., open, in-progress, closed).

### PR DESCRIPTION
**Pull request by disvid**
### Related Issue  
Closes #18 

### Type of Change
Put `x` inside the square bracket to specify what type of change your PR is:  
- [x] New Feature  
- [ ] Bug Fix  
- [ ] Code Refactor  
- [ ] Documentation Update  
- [ ] Other (please specify): 

### Description of Change  
Adds filters to show tickets based on status (e.g., open, in-progress, closed).

### Implementation Details  
required changes have been done in app/page.jsx file.

### Demo  

Screenshot from 2024-12-20 21-33-55
Screenshot from 2024-12-20 21-34-08

### CODE REVIEW

**Comment by iamgoeldhruv** [comment]
@disvid There is mistake in description of issue. Actually it should not be open , inProgress, Closed as there are no such options like these in status. Please update it to( All , Not Started , Started ,Done) as per status field.

**Comment by disvid** [comment]
@iamgoeldhruv Done.

**Comment by AyushDharDubey** [comment]
Hi @disvid, we encourage you trying to fix this issue but it would be nice if contributors would first discuss their idea of solution on the issue page before opening a PR.

Unfortunately, your fix doesn’t seem to resolve the issue. Please make sure to thoroughly test your solution locally to confirm it works before submitting a PR. If you need any hints or guidance regarding why it fails, feel free to reach out, we’re happy to help!

Additionally, please avoid committing undesired changes. For example, altering the `Dashboard` object from an `async` to a `sync` function wasn’t necessary in this case. Keeping your PR clean and focused on the issue makes it easier for everyone to review.

Thanks again for your contribution! We look forward to your next update.

**Comment by disvid** [comment]
@AyushDharDubey really sorry sir, i have made the required changes and the filter is now working.

**Comment by AyushDharDubey** [comment]
@disvid would you be able to resolve the merge conflicts ?

**Comment by iamgoeldhruv** [comment]
@disvid Please resolve conflicts as soon as possible

**Comment by AyushDharDubey** [comment]
Hi @disvid, 
this is to make you aware that if you are unable to respond or resolve the issue by the day after tomorrow, 2359 hours, unfortunately, we will need to close the PR to give other contributors a chance.

if you require more time feel free to let us know. if you need assistance or will not be able to resolve the merge conflict at all, kindly inform us before 28th Dec, 2359 hours so that we can explore alternative solutions...

**Comment by disvid** [comment]
@AyushDharDubey @iamgoeldhruv Sir the mergeconflicts have been resolved.

File: app/page.jsx [hunk 1]
```diff
@@ -78,16 +116,17 @@ const Dashboard = () => {
                     .sort((a, b) => b.priority - a.priority)
                     .map((filteredTicket, ticketIndex) => (
                       <TicketCard
-                        id={ticketIndex}
-                        key={ticketIndex}
+                        id={filteredTicket._id || ticketIndex}
```
Comment (AyushDharDubey) [inline app/page.jsx:119]: i am not able to figure out any edge case where `ticketIndex` might fail

if there are none, please change it to `ticketIndex` only

File: app/page.jsx [hunk 2]
```diff
@@ -17,37 +17,58 @@ const getTickets = async () => {
...
+  }, [statusFilter, tickets]);
+
+  const uniqueCategories = [...new Set(filteredTickets?.map(({ category }) => category))];
   const uniqueStatuses = [...new Set(tickets?.map(({ status }) => status))];
 
   const handleViewByChange = (event) => {
     setViewBy(event.target.value);
   };
 
+  const handleStatusChange = (event) => {
```
Comment (AyushDharDubey) [inline app/page.jsx:59]: please rename this object to handleStatusFilterChange

Comment (disvid) [reply app/page.jsx]: Done

**Comment by AyushDharDubey** [comment]
Hi @disvid,

good work, i can see that the PR is capable of fixing the issue and is ready to be merged.

just one suggestion for you, please make sure that your commit message convey precisely the changes you are committing.
thank you.

**Pull request merged**